### PR TITLE
chore: Azure NPM update to v1.4.16

### PR DESCRIPTION
**Pull request by vakalapa**
**Reason for Change**:

Updating Azure NPM to v1.4.9

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [ ] No
- [x] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [x] Yes

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: parts/k8s/addons/azure-network-policy.yaml [hunk 1]
```diff
@@ -121,3 +121,37 @@ spec:
           path: /etc/protocols
           type: File
       serviceAccountName: azure-npm
+---
+apiVersion: v1
```
Comment (jadarsie) [inline parts/k8s/addons/azure-network-policy.yaml:125]: New resources need label `addonmanager.kubernetes.io/mode`

File: pkg/api/k8s_versions.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ const (
 	antreaAgentImageReference                                = antreaControllerImageReference
 	antreaOVSImageReference                                  = antreaControllerImageReference
 	antreaInstallCNIImageReference                           = antreaControllerImageReference
-	azureNPMContainerImageReference                   string = "mcr.microsoft.com/containernetworking/azure-npm:v1.2.2_hotfix"
+	azureNPMContainerImageReference                   string = "mcr.microsoft.com/containernetworking/azure-npm:v1.4.16"
```
Comment (vakalapa) [inline pkg/api/k8s_versions.go:31]: The latest image we have is v1.4.22, lmk if you want me to make the change or you can continue.

Comment (jadarsie) [reply pkg/api/k8s_versions.go:31]: You are free to move this forward. Just one thing, I noticed that the current `cni` is v1.4.16:
https://github.com/Azure/aks-engine/blob/master/pkg/api/const.go#L322-L329

I am assuming it is better to keep `cni` and `nmp` in sync, but you probably know better how that works.

If you move to a different `nmp` version than v1.4.16 (using v1.4.22 as an example), then we should double check that
https://github.com/Azure/aks-engine/blob/master/parts/k8s/addons/azure-network-policy.yaml
is capturing changes in 
https://github.com/Azure/azure-container-networking/blob/v1.4.22/npm/azure-npm.yaml

**Comment by jackfrancis** [comment]
@jadarsie is there an Azure Stack Hub reason to integrate this change?

**Comment by jadarsie** [comment]
Nope, you can abandon if that's your preference

**Comment by jackfrancis** [comment]
Given upcoming deprecation, we won't do this additional addon integration.

**Pull request closed without merging**